### DISCUSSION TITLE
Disable SFTP password authention for a user

### DISCUSSION THREAD

**Discussion by dkebler** [drakkan/sftpgo#1520] [Q&A] [answered]
In the UI on can add a public key.  

If I don't then it is possible to login with the user password

If I do add one then I can login without password using the private key in my agent.

If I then remove the key from my agent I will be prompted for password (which is bad bad)

I don't understand because I see no way in the UI disable password authentication like I would set in an openssh server config.

I mean I simply add to sshd config.

```
KbdInteractiveAuthentication no
PubkeyAuthentication yes
PasswordAuthentication no
```

so is that not possible in the web ui?  If not how to I configure that via environment variables or directly?

I'm using the latest alpine image.

**Comment by dkebler** [comment]
hmm...

https://sftptogo.com/docs/security/using-ssh-key-authentication

looks like the self hosted images don't have an organization page so no way to set this via a UI.  

Might still be a way to set this directly in some conf file?

**Reply by drakkan** [reply]
this is SFTPGo not SFTP To Go

**Comment by dkebler** [comment]
look like I can set those by creating a file on the host mounted /var/lib/sftpgo volume.

https://github.com/drakkan/sftpgo/blob/main/docs/full-configuration.md

**Comment by dkebler** [comment]
ok I have this.

```
sftpd:
  password_authentication: false
  keyboard_interactive_authentication: false
```

but if I read the docs correctly I put that in  `/var/lib/sftpgo/.config/sftpgo.yaml`  (mounted on my host) and restart the server and nope it is not disabled.    So not sure where to put this file.

```
The configuration can be read from JSON, TOML, YAML, HCL, envfile and Java properties config files. If your config-file flag is set to sftpgo (default value), you need to create a configuration file called sftpgo.json or sftpgo.yaml and so on inside config-dir.
```

**Comment by dkebler** [comment, marked as answer]
Ok I have figured this out

assuming you have `/var/lib/sftpgo` mounted on your host 

```
sftpd:
  password_authentication: false
  keyboard_interactive_authentication: false
```

in  `/var/lib/sftpgo/sftpgo.yaml`

then I had to explicitly set the  `SFTPGO_CONFIG_DIR=/var/lib/sftpgo`  environment variable in the environment file I am including in my docker compose file.

What would be nice is to be able to do this per user instead of for the entire server.  That's something I could do via Open SSH configs using a user match but I don't see any support in the configuration for matches (user or group) so this seems like a "downgrade" using sftpgo. :(

**Comment by drakkan** [comment]
> What would be nice is to be able to do this per user instead of for the entire server. That's something I could do via Open SSH configs using a user match but I don't see any support in the configuration for matches (user or group) so this seems like a "downgrade" using sftpgo. :(

This is supported, check the user page

**Reply by dkebler** [reply]
you mean in the web UI right? I never saw such a setting that's why i went looking elsewhere.  Screenshot please? 

yea sorry about docs confusion.  is the a toc page for that docs subfolder?
